be/src/exec/hash_table.cpp

```diff
@@ -244,7 +244,7 @@ void HashTable::grow_node_array() {
     _alloc_list.push_back(_current_nodes);
     _end_list.push_back(_current_nodes + alloc_size);
 
-    _mem_tracker->Consume(alloc_size);
+    _mem_tracker->consume(alloc_size);
```

Comment:
There are many codes call mem_tracker->somsume(size_t), Is there a better method to do this automatically?
For example, we may rewrite the memory allocators and track the memory usage during allocate

Reply:
@yiguolei  I have done something similar with reference to clickhouse to automatically count memory usage via TCMalloc Hook
PR: https://github.com/apache/incubator-doris/pull/7198
But this PR is too big, so I was advised to split the commit =_=

Different from Jemalloc, overloading the TCMalloc new/delete operator needs to invade the source code of TCMalloc, so it is implemented by Hook;

Reply:
There are too many places call memory tracker and we may lost to add mem_tracker.consume in some node or thread.    If we are following clickhouse's method, I think it do not need to call memory tracker everywhere and call mem tracker.consume so many times.

Reply:
> There are too many places call memory tracker and we may lost to add mem_tracker.consume in some node or thread. If we are following clickhouse's method, I think it do not need to call memory tracker everywhere and call mem tracker.consume so many times.
> 
> We could attach the running thread to a conext, the context maybe a query context or a compaction context or olap scanner. And rewrite the new and delete method to update the memory tracker automatically.

This is done by following next PR.

Reply:
> There are too many places call memory tracker and we may lost to add mem_tracker.consume in some node or thread. If we are following clickhouse's method, I think it do not need to call memory tracker everywhere and call mem tracker.consume so many times.
> 
> We could attach the running thread to a conext, the context maybe a query context or a compaction context or olap scanner. And rewrite the new and delete method to update the memory tracker automatically.

You are right, I have already implemented it like this. In #7198 mentioned above, 
you can see the original design document last year: https://shimo.im/docs/DT6JXDRkdTvdyV3G